lib/Runtime/Library/WasmLibrary.cpp

```diff
@@ -269,14 +254,23 @@ namespace Js
                 ctx->GetOrAddPropertyRecord(funcExport->name, funcExport->nameLength, &propertyRecord);
                 Var funcObj;
                 // todo:: This should not happen, we need to add validation that the `function_bodies` section is present
-                if (funcExport->funcIndex < wasmModule->GetFunctionCount())
+                uint32 funcIndex = funcExport->funcIndex;
+                if (funcIndex >= wasmModule->GetImportCount())
```

Comment:
maybe we could introduce helpers to treat `importFunctions` and `localModuleFunctions` (and their WasmBinaryReader counterparts) as one unified namespace.  That could reduce the number of index adjustments and ifs